### PR TITLE
Gallery bug fix

### PR DESCRIPTION
**Pull request by grizzlytheodore**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/compute/Azure.ResourceManager.Compute/src/Generated/CommunityGalleryImageData.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ internal CommunityGalleryImageData()
...
-        internal CommunityGalleryImageData(string name, AzureLocation? location, ResourceType? resourceType, string uniqueId, SupportedOperatingSystemType? osType, OperatingSystemStateType? osState, DateTimeOffset? endOfLifeOn, GalleryImageIdentifier identifier, RecommendedMachineConfiguration recommended, Disallowed disallowed, HyperVGeneration? hyperVGeneration, IReadOnlyList<GalleryImageFeature> features, ImagePurchasePlan purchasePlan, ArchitectureType? architecture, Uri privacyStatementUri, string eula) : base(name, location, resourceType, uniqueId)
+        internal CommunityGalleryImageData(string name, AzureLocation? location, ResourceType? resourceType, string uniqueId, SupportedOperatingSystemType? osType, OperatingSystemStateType? osState, DateTimeOffset? endOfLifeOn, CommunityGalleryImageIdentifier identifier, RecommendedMachineConfiguration recommended, Disallowed disallowed, HyperVGeneration? hyperVGeneration, IReadOnlyList<GalleryImageFeature> features, ImagePurchasePlan purchasePlan, ArchitectureType? architecture, Uri privacyStatementUri, string eula) : base(name, location, resourceType, uniqueId)
         {
```
Comment (grizzlytheodore) [inline sdk/compute/Azure.ResourceManager.Compute/src/Generated/CommunityGalleryImageData.cs:45]: @ArcturusZhang , how do i mitigate this breaking change ?

File: sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs [hunk 2]
```diff
@@ -413,7 +413,7 @@ public partial class CommunityGalleryImageData : Azure.ResourceManager.Compute.M
         public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.Compute.Models.GalleryImageFeature> Features { get { throw null; } }
         public Azure.ResourceManager.Compute.Models.HyperVGeneration? HyperVGeneration { get { throw null; } }
         public Azure.Core.ResourceIdentifier Id { get { throw null; } }
-        public Azure.ResourceManager.Compute.Models.GalleryImageIdentifier Identifier { get { throw null; } }
+        public Azure.ResourceManager.Compute.Models.CommunityGalleryImageIdentifier Identifier { get { throw null; } }
```
Comment (ArcturusZhang) [inline sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs:416]: Well looks like this change is unnecessary for dotnet as well - this property does not have a setter, therefore the setters and the full parameter ctor on `GalleryImageIdentifier` are meaningless for this property. Therefore if we do not change this, it works as before perfectly.
Therefore this is definitely an unnecessary change for dotnet

Comment (ArcturusZhang) [reply sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs:416]: well after some thoughts I believe our previous might not work if the response returns null value for any of these properties - we might get NRE during the deserialization because the previous model only have one constructor and it is expecting all those values are not null.
But still I think this change in the swagger is an overkill, we do not need to introduce such a big change - could we remove the required flags on `GalleryImageIdentifier` model?

Comment (grizzlytheodore) [reply sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs:416]: please see my email on what I think creating this new model is necessary @ArcturusZhang

**Review by ArcturusZhang — APPROVED** [review]
